### PR TITLE
Run tests with Go 1.21

### PR DESCRIPTION
**Pull request by adamantike**
What does this PR change? What problem does it solve?
-----------------------------------------------------

Run tests with Go 1.21.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -22,6 +22,28 @@ jobs:
...
+          - job_name: Windows
+            go: 1.21.x
+            os: windows-latest
+
+          - job_name: macOS
+            go: 1.21.x
+            os: macOS-latest
+            test_fuse: false
+
+          - job_name: Linux
```
Comment (MichaelEischer) [inline .github/workflows/tests.yml:34]: We only keep the full set of tests for the latest go version, as that version will be used for the release. Keeping the full set of tests for older go versions increases the CI runtime too much.

Comment (adamantike) [reply .github/workflows/tests.yml:34]: So, should I just maintain `Linux` for 1.20.x (which also runs for 1.18 and 1.19), to end up with the following matrix...?
* `1.21.x` for Windows, MacOS, Linux, and Linux (race)
*  `1.20.x` for Linux
*  `1.19.x` for Linux
*  `1.18.x` for Linux

Comment (MichaelEischer) [reply .github/workflows/tests.yml:34]: Yes. Just to be clear, the `test_cloud_backends:` should only run for Linux + Go 1.21.x

**Review by MichaelEischer — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**